include/mfmg/dealii/amge_host.templates.hpp

```diff
@@ -76,10 +48,48 @@ struct NoOp
     std::ignore = src;
     std::ignore = dst;
     // Raise an error to make sure nobody uses this by inadvertance
-    throw std::runtime_error("should never get here");
+    throw std::logic_error("should never get here");
   }
 };
-#endif
+
+struct NotAMatrix
+{
+  using MatrixType = dealii::SparseMatrix<double>;
+  using SizeType = MatrixType::size_type;
+
+  template <typename MeshEvaluator, typename DoFHandler>
+  NotAMatrix(MeshEvaluator const &mesh_evaluator, DoFHandler &dof_handler,
+             dealii::AffineConstraints<double> &constraints)
+  {
+    mesh_evaluator.evaluate_agglomerate(dof_handler, constraints,
+                                        sparsity_pattern, _matrix);
```

Comment:
Assert `_matrix.m() == _matrix.n()`?

Reply:
Nah.  Will be caught by the eigensolver if it requires it.

Reply:
But it may segfault somewhere when you get diagonal values and your matrix is tall and skinny (though I do not envision this use case) as you use `m()` when getting diagonal values.